.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,46 @@
+name: Build and Push
+
+on:
+  push:
+    branches:
+      - '*'
```

Comment:
I used `*` because of from travis CI builds noted that we are running all the branches in upstream to build and push images to the respective branch-tag https://quay.io/repository/project-codeflare/mcad-controller?tab=tags .. If its not required we can keep this to only `main`

Reply:
@z103cb ^

Reply:
I think we need to accommodate "patch" a release branch use case, @Srihari1192 will your code handle that use cases? (it seems that it would).

Reply:
Changed to **main** but wondering whether we need to push images for branches other than main aswell

Reply:
Thinking what cases should we cover here:
New commit on `main` branch - new `dev` image published
New commit on `release-<version>` branch - new `dev-release-<version>` image published?
Any other case for producing non release image?

Those ^ are just my ideas, feel free to share your ideas.

Reply:
@sutaakar I think you have covered those. Is there any documentation how to "patch" a release? if we do that is great, if not we should create a separate issue / pr to create it. 

I am Ok with restricting the branches to only the patterns you have outlined.

Reply:
I don't see any specific mention about patch release in readme. I would say it follows the similar flow as a normal release:
- Create a `release-<version>` branch if not exists
- Merge code changes until ready for patch release
- Trigger a release workflow, specify `release-<version>` branch as a workflow source
Ideally this should build proper image and publish correct image tag, though I haven't tried it by myself.

Reply:
@sutaakar @z103cb I think we can put condition for assigning tag value like this. so that in case of main branch and release branch image will be pushed to the respective tag .. wdyt?
```

` if [[ "$GIT_BRANCH"="main" ]]; then
             echo "TAG=dev" >> $GITHUB_ENV
          elif [[ "$GIT_BRANCH" == release-* ]]; then
            echo "TAG=$GIT_BRANCH" >> $GITHUB_ENV
          else
            echo "TAG=$GIT_BRANCH-$(git describe --abbrev=0 --tags)" >> $GITHUB_ENV`
```

Reply:
in principle fine for me
just be careful on content of `GIT_BRANCH`, AFAIK for branch it is like `refs/heads/<branch name>`, see https://github.com/project-codeflare/codeflare-operator/blob/2a4731a237cf1996e6370def8c3dc7842fc2db52/.github/workflows/tag-and-build.yml#L150
Also I think that ` release-<branch name>` format is used for released images, so you need some other name to distinguish it (i.e. like `dev-release-<branch name>`)

Reply:
Okay for release branch will update tag as dev-<release-branch>